src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -57,6 +57,8 @@ protected SourceMethodSymbolWithAttributes(SyntaxReference syntaxReferenceOpt)
                 case RecordDeclarationSyntax recordDecl:
                     Debug.Assert(recordDecl.IsKind(SyntaxKind.RecordDeclaration));
                     return recordDecl;
+                case ClassDeclarationSyntax classDecl:
+                    return classDecl;
```

Comment:
Where is this used? (Which tests does it affect?)

Reply:
> Where is this used? (Which tests does it affect?)

Today this code path is not used, and for records as well. However its behavior is consistent with what SynthesizedPrimaryConstructor.TryGetBodyBinder is doing